Service/AtendimentoService.php

```diff
@@ -461,6 +444,33 @@ public function distribuiSenha($unidade, $usuario, $servico, $prioridade, Client
         $this->dispatcher->createAndDispatch('attending.pre-create', [$atendimento], true);
         
         try {
+            /**
+             * MySQL memory autoincrement fix
+             */
+            $atCount = $om
+                ->getRepository(Atendimento::class)
+                ->createQueryBuilder('a')
+                ->select("COUNT(a.id)")
+                ->getQuery()
+                ->getSingleScalarResult();
+            if ($atCount == 0) {
+                // Count the attendances history
+                $atHistCount = $om
+                    ->getRepository(AtendimentoHistorico::class)
+                    ->createQueryBuilder('ah')
+                    ->select("COUNT(ah.id)")
+                    ->getQuery()
+                    ->getSingleScalarResult();
+
+                if ($atHistCount > 0) {
+                    // If it has attendance history and attendance table is empty
+                    // Set the auto_increment based on history attendance.
+                    $conn = $om->getConnection();
+                    $atHistCount++;
+                    $stmt = $conn->prepare("ALTER TABLE atendimentos AUTO_INCREMENT = " . $atHistCount);
+                    $stmt->execute();
+                }
+            }
```

Comment:
The purpose of this service class is to be agnostic. The right place to put MySQL specific code is on the `MySQLStorage` implementation:

https://github.com/novosga/novosga/blob/master/src/Infrastructure/Storage/MySQLStorage.php